tests/framework/extensions/charts/rancheristio.go

```diff
@@ -0,0 +1,267 @@
+	istioValues := map[string]interface{}{
+		"tracing": map[string]interface{}{
+			"enabled": rancherIstioOpts.Tracing,
+		},
+		"kiali": map[string]interface{}{
+			"enabled": rancherIstioOpts.Kiali,
+		},
+		"ingressGateways": map[string]interface{}{
+			"enabled": rancherIstioOpts.IngressGateways,
+		},
+		"egressGateways": map[string]interface{}{
+			"enabled": rancherIstioOpts.EgressGateways,
+		},
+		"pilot": map[string]interface{}{
+			"enabled": rancherIstioOpts.Pilot,
+		},
+		"telemetry": map[string]interface{}{
+			"enabled": rancherIstioOpts.Telemetry,
+		},
+		"cni": map[string]interface{}{
+			"enabled": rancherIstioOpts.CNI,
+		},
+	}
+	chartInstall := newChartInstall(p.Name, p.InstallOptions.Version, p.InstallOptions.ClusterID, p.InstallOptions.ClusterName, p.Host, istioValues)
+	chartInstalls := []types.ChartInstall{*chartInstall}
+
+	chartInstallAction := newChartInstallAction(p.Namespace, p.InstallOptions.ProjectID, chartInstalls)
+
+	return chartInstallAction
+}
+
+func UpgradeRancherIstioChart(client *rancher.Client, installOptions *InstallOptions, rancherIstioOpts *RancherIstioOpts) error {
+	istioChartUpgradeActionPayload := &payloadOpts{
+		InstallOptions: *installOptions,
+		Name:           RancherIstioName,
+		Host:           client.RancherConfig.Host,
+		Namespace:      RancherIstioNamespace,
+	}
+
+	chartUpgradeAction := newIstioChartUpgradeAction(istioChartUpgradeActionPayload, rancherIstioOpts)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	err = catalogClient.UpgradeChart(chartUpgradeAction)
+	if err != nil {
+		return err
+	}
+
+	// wait for chart to be in status pending upgrade
+	watchAppInterface, err := catalogClient.Apps(RancherIstioNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherIstioName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusPendingUpgrade) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// wait for chart to be full deployed
+	watchAppInterface, err = catalogClient.Apps(RancherIstioNamespace).Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + RancherIstioName,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+		app := event.Object.(*catalogv1.App)
+
+		state := app.Status.Summary.State
+		if state == string(catalogv1.StatusDeployed) {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
Shouldn't all this block be part of the UpgradeChart function by passing the App namespace too?